src/main/java/dev/qixils/fahare/Fahare.java

```diff
@@ -70,6 +72,14 @@ public final class Fahare extends JavaPlugin implements Listener {
         return world;
     }
 
+    private @NotNull World createFakeOverworldSetSeed() {
+        // Create fake overworld
+        WorldCreator creator = new WorldCreator(fakeOverworldKey).copy(overworld()).seed(Seed);
+        World world = Objects.requireNonNull(creator.createWorld(), "Could not load fake overworld");
+        world.setDifficulty(overworld().getDifficulty());
+        return world;
```

Comment:
silly code repetition with `createFakeOverworld`.  
Could be a single function with a seed argument that is a random seed for the no-set-seed case